src/main/java/com/microsoft/cse/helium/app/controllers/Controller.java

```diff
@@ -0,0 +1,96 @@
+package com.microsoft.cse.helium.app.controllers;
+
+import com.microsoft.cse.helium.app.Constants;
+import com.microsoft.cse.helium.app.dao.ActorsDao;
+import com.microsoft.cse.helium.app.dao.MoviesDao;
+import com.microsoft.cse.helium.app.models.Entity;
+import com.microsoft.cse.helium.app.utils.ParameterValidator;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+@Component
+public class Controller implements IController {
+
+  @Autowired ActorsDao actorsDao;
+  @Autowired MoviesDao movieDao;
+
+  @Autowired ParameterValidator validator;
+
+  private static final Logger logger = LoggerFactory.getLogger(Controller.class);
+
+  /** commonControllerUtilAll. */
+  public Object getAll(
```

Comment:
Should this be protected so that it is only visible by children?

Reply:
https://stackoverflow.com/questions/5376970/protected-in-interfaces